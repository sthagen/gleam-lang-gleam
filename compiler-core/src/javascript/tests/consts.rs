use crate::assert_js;

#[test]
fn custom_type_constructor_imported_and_aliased() {
    assert_js!(
        ("package", "other_module", "pub type T { A }"),
        r#"import other_module.{A as B}

pub const local = B
"#,
    );
}

#[test]
fn imported_aliased_ok() {
    assert_js!(
        r#"import gleam.{Ok as Y}

pub type X {
  Ok
}

pub const y = Y
"#,
    );
}

#[test]
fn imported_ok() {
    assert_js!(
        r#"import gleam

pub type X {
  Ok
}

pub const y = gleam.Ok
"#,
    );
}
