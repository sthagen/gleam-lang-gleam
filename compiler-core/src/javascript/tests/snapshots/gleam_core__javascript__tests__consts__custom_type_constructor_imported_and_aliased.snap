---
source: compiler-core/src/javascript/tests/consts.rs
assertion_line: 5
expression: "import other_module.{A as B}\n\npub const local = B\n"
---
import * as $other_module from "../../package/other_module.mjs";
import { A as B } from "../../package/other_module.mjs";

export const local = new B();
